lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -15,6 +16,10 @@ use tracing::{debug, warn};
 use vector_config::configurable_component;
 
 const GELF_MAGIC: &[u8] = &[0x1e, 0x0f];
+const GZIP_MAGIC: &[u8] = &[0x1f, 0x8b];
```

Comment:
Should we add other decompression types such as zstd,lz4 and etc? 

Other gelf decoders (https://github.com/grafana/go-gelf/blob/25db8704bcf3f484c958312cd0cc49e5c768dcf1/gelf/reader.go#L120 and  // and https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L157) takes only gzip and zlib into account. This is what the GELF documentation also states https://go2docs.graylog.org/current/getting_in_log_data/gelf.html

Reply:
We can always add more later. It's good as is. 

Tangential, TIOLI:
I am a bit worried because are repeating the magic bytes across the codebase: [usages](https://github.com/search?q=repo%3Avectordotdev%2Fvector%20%22is_gzip%22&type=code). If we can at least reuse the constant you are introducing here, it would be great.

Reply:
addressed in https://github.com/vectordotdev/vector/pull/21816/commits/3e4416f7a8972433a696d7bc39b48efaeaf8eba3